### PR TITLE
cfitsio: 3.430 -> 3.450

### PR DESCRIPTION
**Pull request by xbreak**
###### Motivation for this change

- Update cfits version from 3.430 to 3.450. Please also note that this update contains security fixes made in 3.440 ([ChangeLog](https://heasarc.gsfc.nasa.gov/FTP/software/fitsio/c/docs/changes.txt)).
- Switch src.url to use https (HEASARC will disable FTP as of September 2019, https://heasarc.gsfc.nasa.gov/docs/FTPWarning.html).
- Added optional bzip2 support in derivation (enabled by default as suggested by @c0bw3b). 

###### Changes from Review
- ~~This also includes a fixup of the `cfitsio.pc` file to point to the store path of the bzip2 libraries.~~ Not needed if bzip2.out is used instead.
- Enabled bzip2 support by default.
- Minor cleanup.
- Added myself as maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (CentOS 7.4)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] *Tested compilation of all pkgs that depend on this change manually
- [x] Tested compilation with bzip support enabled and manually inspected libraries and pc-file.
- [x] **Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
---
\* I had to manually build some packages due to error cloning builder process (I'm running single user on CentOS without sandboxing).
\** Tested to load fits files with giv

Built ok:
- cfitsio
- gimpPlugins.ufraw
- ufraw
- indilib
- gildas
- gimp-with-plugins
- giv

### CODE REVIEW

**Review by tomberek — APPROVED** [review]
Tested on Ubuntu.

**Comment by c0bw3b** [comment]
Hello @xbreak and welcome around here :)

>  ** There's no maintainer, can I add one?

Yes, definitely! You can add yourself to the maintainer list if you wish to maintain this package.
This should be done in a separate commit. See PR 61137 for a recent example.

File: pkgs/development/libraries/cfitsio/default.nix [hunk 1]
```diff
@@ -1,19 +1,32 @@
-{ fetchurl, stdenv }:
+{ fetchurl, stdenv
 
- stdenv.mkDerivation {
-  name = "cfitsio-3.430";
+# Optional dependencies
+, bzip2 ? null }:
```
Comment (c0bw3b) [inline pkgs/development/libraries/cfitsio/default.nix]: I guess we can build this _with_ bz2 support by default.
Tested locally and there is no noticeable increase in closure size.

File: pkgs/development/libraries/cfitsio/default.nix [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
-    sha256 = "07fghxh5fl8nqk3q0dh8rvc83npnm0hisxzcj16a6r7gj5pmp40l";
+    url = "ftp://heasarc.gsfc.nasa.gov/software/fitsio/c/cfitsio${builtins.replaceStrings ["."] [""] version}.tar.gz";
+    sha256 = "0bmrkw6w65zb0k3mszaaqy1f4zjm2hl7njww74nb5v38wvdi4q5z";
   };
 
+  buildInputs = [ bzip2 ];
+
   patches = [ ./darwin-curl-config.patch ./darwin-rpath-universal.patch ];
 
+  configureFlags = stdenv.lib.optional (bzip2 != null) "--with-bzip2=${bzip2.dev}";
```
Comment (c0bw3b) [inline pkgs/development/libraries/cfitsio/default.nix]:
```suggestion
  configureFlags = stdenv.lib.optional (bzip2 != null) "--with-bzip2=${bzip2.out}";
```

Then you wouldn't need the postFixup fix and libcfitsio is still linked to the correct libbz2
```
$ ldd result/lib/libcfitsio.so
[...]
        libbz2.so.1 => /nix/store/wlivxbc4ny21gzq9a561bwx9b4p08m4m-bzip2-1.0.6.0.1/lib/libbz2.so.1 (0x00007f331f773000)
```

File: pkgs/development/libraries/cfitsio/default.nix [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
   # Shared-only build
   buildFlags = "shared";
   postPatch = '' sed -e '/^install:/s/libcfitsio.a //' -e 's@/bin/@@g' -i Makefile.in
-   '';
+   '' ;
```
Comment (c0bw3b) [inline pkgs/development/libraries/cfitsio/default.nix]:
```suggestion
   '';
```
I guess that was not intended ^^

**Review by c0bw3b — COMMENTED** [review]
Can you also clarify the license in meta? This is MIT so 
```nix
  license = licenses.mit;
```
and the comment about BSD-style licensing can be removed

File: pkgs/development/libraries/cfitsio/default.nix (hunk 1 above)
Comment (xbreak) [reply pkgs/development/libraries/cfitsio/default.nix]: Ok!

File: pkgs/development/libraries/cfitsio/default.nix (hunk 2 above)
Comment (xbreak) [reply pkgs/development/libraries/cfitsio/default.nix]: Much better, thanks! I didn't realize the bzip2 headers could be resolved via `NIX_CFLAGS_COMPILE`, which was why I used `bzip2.dev`.

File: pkgs/development/libraries/cfitsio/default.nix (hunk 3 above)
Comment (xbreak) [reply pkgs/development/libraries/cfitsio/default.nix]: Thanks, will fix!

**Comment by xbreak** [comment]
@c0bw3b: Thanks for the review. I've updated the PR with the requested changes. Apart from your comments I also changed the url to use https since HEASARC are phasing out unencrypted FTP access.

**Review by c0bw3b — APPROVED** [review]

**Comment by c0bw3b** [comment]
Perfect. Thanks for your contribution.

**Comment by c0bw3b** [comment]
Forgot to mention: wrt security fixes, the grave public vulnerabilities (4 CVE published) were already fixed in v3.43 that we ship in release 19.03

Now 3.44 may have cleaned up some more codebase but it's undocumented what was fixed exactly and there is no CVE. So exploitability is very close to null.
For this reason I think a backport to stable is not warranted here.

**Pull request merged**